### PR TITLE
feat(gs): mute installations app-wide from the Cluster access popover

### PR DESCRIPTION
**Pull request by marians**
### What does this PR do?

Adds a per-installation **on/off toggle** to the sidebar **Cluster access** popover. A *muted* installation is switched off for the whole app:

- the `ClusterAccessConnector` stops probing it (and drops it from the access-status set);
- fleet views stop fetching from it — the shared `InstallationPicker` folds the muted set into `disabledClusters` (which `MultipleClustersSelector` already excludes from the active/fetched set), covering the **Clusters** and **Deployments** pages; and the **agent-platform** lists exclude it automatically (muted → not probed → not `healthy` → not reachable).

The preference is user intent, so it is persisted per-browser (`localStorage`, key `gs-muted-installations`) and synced across tabs — deliberately distinct from the health-based `useDisabledInstallations` hook, which auto-detects unreachable backends.

Implemented as a new `MutedInstallationsApi` / `MutedInstallationsStore` (mirrors the cluster-access status store, but persisted), registered in the gs plugin and exported. The toggle itself is a bui (`@backstage/ui`) `Switch`.

Also fixes a pre-existing quirk surfaced by the toggle: a successful broker cluster-token mint recorded the installation as **healthy** (green) even when the apiserver was unreachable, so un-muting an unreachable installation flashed green before the `/version` probe corrected it to degraded. Health is now owned solely by the authoritative `/version` probe; the mint still records the precise broker-level failure states (session-expired / degraded), which the probe defers to.

### What is the effect of this change to users?

Users can turn individual installations off from the Cluster access popover. A muted installation is no longer probed and no longer appears / is fetched in the Clusters, Deployments, or Agent Platform views. It shows as an "Off" row in the popover and can be switched back on at any time. The setting persists across reloads. Un-muting an unreachable installation now shows connecting → degraded, with no misleading green flash (this also removes the brief false-green during initial app load).

### How does it look like?

image

### Any background context you can provide?

Follow-up to the agent-platform reachability work (#1957). The toggle deliberately lives in the shared Cluster access widget rather than the agent-platform page so it applies app-wide.

### Do the docs need to be updated?

No.

### Should this change be mentioned in the release notes?

- [ ] A changeset describing the change and affected packages was added. (`@giantswarm/backstage-plugin-gs` is `private: true`, not published to npm — no changeset needed; released via the app image/chart auto-release.)

### CODE REVIEW

File: plugins/gs/src/components/ClusterAccessStatus/ClusterAccessConnector.tsx [hunk 1]
```diff
@@ -121,9 +122,30 @@ export function ClusterAccessConnector() {
...
+  // Track the user's muted set reactively so muting/unmuting re-runs the probe
+  // effects (a muted installation must stop being probed and drop off the
+  // status set; unmuting must resume probing). The store emits only on real
+  // changes, so `muted`'s identity is stable between changes.
+  const [muted, setMuted] = useState<string[]>(() => mutedApi.getSnapshot());
   useEffect(() => {
-    const installations = authProvidersApi.getBrokerCoveredInstallations();
+    const subscription = mutedApi.muted$().subscribe(setMuted);
+    return () => subscription.unsubscribe();
+  }, [mutedApi]);
```
Comment (marians) [inline plugins/gs/src/components/ClusterAccessStatus/ClusterAccessConnector.tsx]: **Startup double-probe: the initial `muted$` replay restarts the whole fleet probe.** The comment (l.127–130) says "the store emits only on real changes, so `muted`'s identity is stable between changes" — but `muted$()` replays `getSnapshot()` on subscribe, and `getSnapshot()` returns a **new array reference** every call (`[...this.muted].sort(...)`). So on mount: `useState(() => mutedApi.getSnapshot())` seeds array A, the effect runs and `connectAll()` starts probing the whole fleet; then the subscription immediately fires `setMuted(getSnapshot())` = array B (same contents, new ref). Since `muted` is a dependency of both probe effects (l.235, l.283) and `B !== A`, React tears down and re-runs them, cancelling the in-flight probes and calling `connectAll()` a second time. Given how much care the probe path takes to stay cheap (`PROBE_TIMEOUT_MS`, bounded concurrency), doubling the startup fleet probe is a real regression. Consider comparing snapshots by content (skip `setMuted` when the sorted arrays are equal) or gating the effect on a stable key like `muted.join('\n')`.

File: plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+export class MutedInstallationsStore implements MutedInstallationsApi {
+  private muted: Set<string>;
+  private readonly subscribers = new Set<(snapshot: string[]) => void>();
+
+  private constructor() {
+    this.muted = MutedInstallationsStore.load();
+    // Keep multiple tabs consistent: another tab writing the key reloads ours.
+    if (typeof window !== 'undefined') {
+      window.addEventListener('storage', event => {
+        if (event.key === STORAGE_KEY) {
```
Comment (marians) [inline plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts]: **Cross-tab sync misses `localStorage.clear()`.** The handler only reloads when `event.key === STORAGE_KEY`, but the `storage` event fires with `key === null` when another tab calls `localStorage.clear()` (and some browsers also use `null` for bulk changes). In that case this tab keeps its stale muted set instead of resetting, so the "synced across tabs" guarantee has a hole. Consider also reloading when `event.key === null` (treat it as "storage changed, re-read our key").

File: plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ */
+export class MutedInstallationsStore implements MutedInstallationsApi {
+  private muted: Set<string>;
+  private readonly subscribers = new Set<(snapshot: string[]) => void>();
+
+  private constructor() {
+    this.muted = MutedInstallationsStore.load();
+    // Keep multiple tabs consistent: another tab writing the key reloads ours.
+    if (typeof window !== 'undefined') {
+      window.addEventListener('storage', event => {
```
Comment (marians) [inline plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts:25]: **`storage` listener is never removed.** The constructor adds a `window` `storage` listener with no corresponding `removeEventListener`. In production the store is a process-wide singleton so this is harmless, but every `MutedInstallationsStore.create()` in the test suite (each `renderSidebar` / `InstallationPicker` test) adds another permanent listener to the shared jsdom `window`. The `MutedInstallationsStore.test.ts` cross-tab test's `dispatchEvent` then fans out to every store created by earlier tests. It's benign today only because `beforeEach` clears `localStorage`, but it's a latent cross-test coupling. Worth noting since the store is otherwise cleanly unit-testable.

File: plugins/gs/src/components/ClusterAccessStatus/ClusterAccessStatusSidebarItem.tsx [hunk 4]
```diff
@@ -151,26 +163,68 @@ function StatusDot({ color }: { color: string }) {
...
+        installation,
+        dotColor: STATE_COLORS[entry.state],
+        caption,
+        enabled: true,
+      };
+    });
+  }, [entries, muted]);
+
+  // Nothing accessed and nothing muted -- keep the sidebar uncluttered.
+  if (entries.length === 0 && muted.length === 0) {
```
Comment (marians) [inline plugins/gs/src/components/ClusterAccessStatus/ClusterAccessStatusSidebarItem.tsx:232]: **All-muted shows a green badge reading "All 0 healthy".** The popover now renders when `entries.length === 0 && muted.length > 0`, but `summarize(entries)` and `overallState(entries)` still ignore `muted`. If the user mutes every installation, `entries` is empty, so `summarize([])` hits `counts.healthy === total` with `total === 0` and returns `[{ label: 'All 0 healthy' }]`, and `overallState([])` falls through to `'healthy'` → a green badge. So a fully-switched-off state advertises "All 0 healthy" in green, the opposite of reality. Consider special-casing `total === 0` (neutral badge / "All off" summary) when there are muted rows but no live entries.

**Review by marians — COMMENTED** [review]
Code review (medium effort, correctness-focused). 4 findings posted inline — one efficiency regression (startup double-probe), one cosmetic-correctness (all-muted green badge), one cross-tab edge case, and one test-coupling note. No blocking bugs found; the core mute/unmute flow and the `disabledClusters` union look correct.

File: plugins/gs/src/components/ClusterAccessStatus/ClusterAccessConnector.tsx (hunk 1 above)
Comment (marians) [reply plugins/gs/src/components/ClusterAccessStatus/ClusterAccessConnector.tsx]: Good catch — fixed in 95602d89. Added a shared `useMutedInstallations()` hook that returns a **content-stable** array reference (it returns the previous array when the sorted contents are unchanged), so the initial `muted$()` replay no longer changes `muted`'s identity and the probe effects don't re-run / restart the fleet probe on mount. The connector, sidebar widget, and installation picker all use it now (also de-duplicates the three copies of the subscribe-to-muted logic).

File: plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts (hunk 2 above)
Comment (marians) [reply plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts]: Fixed in 95602d89 — the `storage` handler now reloads on `event.key === STORAGE_KEY || event.key === null`, so another tab's `localStorage.clear()` (which fires with `key === null`) resets this tab's muted set. Added a test covering the `key === null` case.

File: plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts (hunk 3 above)
Comment (marians) [reply plugins/gs/src/apis/mutedInstallations/MutedInstallationsStore.ts:25]: Acknowledged — leaving as-is intentionally. In production the store is an app-lifetime singleton (one instance via the `ApiBlueprint` factory), so the listener living for the app's lifetime is correct and there's no natural teardown hook. The cross-test coupling is real but benign: `beforeEach` clears `localStorage`, so a fanned-out `storage` event just reloads every store to the same (cleared/seeded) value. Adding a `dispose()` purely for test hygiene would widen the API for no production benefit, so I'd rather not.

File: plugins/gs/src/components/ClusterAccessStatus/ClusterAccessStatusSidebarItem.tsx (hunk 4 above)
Comment (marians) [reply plugins/gs/src/components/ClusterAccessStatus/ClusterAccessStatusSidebarItem.tsx:232]: Fixed in 95602d89. When there are no live entries but some muted installations, the badge now uses a neutral grey (`MUTED_COLOR`) instead of green, and the summary shows "N off" instead of "All 0 healthy". `summarize`/`overallState` are unchanged (still entries-only); the muted case is handled in the component. Added a test asserting the all-muted state shows "N off" and not "All 0 healthy".
